File: onebot/src/main/java/cn/evole/onebot/sdk/response/guild/GuildMemberProfileResp.java
package cn.evole.onebot.sdk.response.guild;

import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.List;

/**
 * @author cnlimiter
 */
@Data
public class GuildMemberProfileResp {

    @SerializedName("tiny_id")
    public String tinyId;

    @SerializedName("nickname")
    public String nickname;

    @SerializedName("avatar_url")
    public String avatarUrl;

    @SerializedName("join_time")
    public String joinTime;

    @SerializedName("roles")
    public List<RoleInfo> roles;

    @Data
    public static class RoleInfo {

        @SerializedName("role_id")
        public String roleId;

        @SerializedName("role_name")
        public String roleName;

    }

}

File: onebot/src/main/kotlin/sdk/response/guild/GuildMemberProfileResp.kt
package cn.evolvefield.onebot.sdk.response.guild

import com.google.gson.annotations.SerializedName
import lombok.Data

@Data
class GuildMemberProfileResp {
    @SerializedName("tiny_id")
    var tinyId: String = ""

    @SerializedName("nickname")
    var nickname: String = ""

    @SerializedName("avatar_url")
    var avatarUrl: String = ""

    @SerializedName("join_time")
    var joinTime: String = ""

    @SerializedName("roles")
    var roles: List<RoleInfo> = mutableListOf()

    @Data
    class RoleInfo {
        @SerializedName("role_id")
        var roleId = ""

        @SerializedName("role_name")
        var roleName = ""
    }
}
